src/mpi/coll/allgather/allgather.c

```diff
@@ -475,9 +475,14 @@ int MPI_Allgather(const void *sendbuf, int sendcount, MPI_Datatype sendtype,
 
     /* ... body of routine ...  */
 
-    mpi_errno = MPID_Allgather(sendbuf, sendcount, sendtype,
-                               recvbuf, recvcount, recvtype,
-                               comm_ptr, &errflag);
+    if (MPIR_CVAR_USE_MPIR_COLLECTIVES)
+        mpi_errno = MPIR_Allgather(sendbuf, sendcount, sendtype,
+                                   recvbuf, recvcount, recvtype,
+                                   comm_ptr, &errflag);
+    else
+        mpi_errno = MPID_Allgather(sendbuf, sendcount, sendtype,
+                                   recvbuf, recvcount, recvtype,
+                                   comm_ptr, &errflag);
```

Comment:
This patch can just get dropped in favor of the similar patch that will go into the prereqs.